python/dgl/graphbolt/sampled_sub_graph.py

```diff
@@ -0,0 +1,172 @@
+"""Heterogeneous sampled subgraph."""
+# pylint: disable= invalid-name
+from collections.abc import Mapping
+from typing import Dict, Tuple, Union
+
+import torch
+
+
+class SampledSubGraph:
+    r"""Class for sampled subgraph."""
+
+    def __init__(
+        self,
+        node_pairs: Union[
+            Dict[Tuple[str, str, str], Tuple[torch.tensor, torch.tensor]],
+            Tuple[torch.tensor, torch.tensor],
+        ],
+        reverse_column_node_ids: Union[
+            Dict[str, torch.tensor], torch.tensor
+        ] = None,
+        reverse_row_node_ids: Union[
+            Dict[str, torch.tensor], torch.tensor
+        ] = None,
+        reverse_edge_ids: Union[
+            Dict[Tuple[str, str, str], torch.tensor], torch.tensor
+        ] = None,
+    ):
+        """Initialize the SampledSubGraph object. In the context of a
+        heterogeneous graph, each field should be of `Dict` type. For
+        homogeneous graphs, each field should correspond to its respective
+        value type.
+
+        Parameters
+        ----------
+        node_pairs :  Tuple[Tensor, Tensor] or Dict[(str, str, str), Tuple[Tensor, Tensor]]
+            Node pairs representing source-destination edges.
```

Comment:
Would `SampleSubgraph` support subgraphs of a multigraph?  If so, how do we distinguish multi-edges for a subgraph from a multigraph if we only use node pairs?  If not, are we going to make a new class to support it?

Reply:
Yes, we support multigraph as well, there is a optional "reverse_edge_ids" that can be used to distinguish multiple edges.